app/main/views/tour.py

```diff
@@ -0,0 +1,141 @@
+from flask import abort, redirect, render_template, session
+
+from app import current_service, current_user, url_for
+from app.main import main
+from app.main.views.send import (
+    all_placeholders_in_session,
+    fields_to_fill_in,
+    get_normalised_placeholders_from_session,
+    get_placeholder_form_instance,
+    get_recipient_and_placeholders_from_session,
+)
+from app.utils import get_template, user_has_permissions
+
+
+@main.route("/services/<uuid:service_id>/tour/<uuid:template_id>")
+@user_has_permissions('send_messages')
+def begin_tour(service_id, template_id):
+    db_template = current_service.get_template_with_user_permission_or_403(template_id, current_user)
+
+    if db_template['template_type'] != 'sms':
+        abort(404)
+
+    template = get_template(
+        db_template,
+        current_service,
+        show_recipient=True,
+    )
+
+    template.values = {"phone_number": current_user.mobile_number}
+
+    session['placeholders'] = {}
+
+    return render_template(
+        'views/templates/start-tour.html',
+        template=template,
+        help='1',
+        continue_link=url_for('.tour_step', service_id=service_id, template_id=template_id, step_index=1)
+    )
+
+
+@main.route(
+    "/services/<uuid:service_id>/tour/<uuid:template_id>/step-<int:step_index>",
+    methods=['GET', 'POST'],
+)
+@user_has_permissions('send_messages', restrict_admin_usage=True)
+def tour_step(service_id, template_id, step_index):
+    db_template = current_service.get_template_with_user_permission_or_403(template_id, current_user)
+
+    if db_template['template_type'] != 'sms':
+        abort(404)
+
+    template = get_template(
+        db_template,
+        current_service,
+        show_recipient=True,
+    )
+
+    placeholders = fields_to_fill_in(template, prefill_current_user=True)
+    try:
+        # user urls are 1 indexed, so start at step-1
+        current_placeholder = placeholders[step_index - 1]
```

Comment:
If someone visits `step-0` this will evaluate to `placeholders[-1]` and show them the last step. We probably don't want this to happen. Any suggestions for how to code this up?

Reply:
I would `404` it because it will never be a valid URL
```suggestion
        if step_index < 1:
            abort(404)
        current_placeholder = placeholders[step_index - 1]
```

Reply:
So the suggestion is to 404 for `step-0` but on something with 3 placeholders, `step-4` would take them back to `step-1`.  Feels slightly inconsistent but I guess it doesn't really matter that much?

Reply:
Part of me feels like we are trying to be too smart by if they try and visit a url for a step doesn't exist, send them to a more appropriate page (like the first step or the page to check the notification). I guess it could be useful but is the code worth it compared to we could just be 404ing always those types of requests.

Reply:
If you have 3 placeholders then `…/step-3` will always redirect to `…/step4`. Then this code notices that `4` is out of bounds, and tries to send them to the `…/check` page. If the some of the placeholders are missing then it catches that, and sends them back to the start, rather than the check page.

I think it’s sensible behaviour to keep for the one-off journey, because someone could edit a template and change the number of placeholders while it’s being sent. So I guess my argument would be to keep it working the same way here for consistency.

Reply:
It appears that that isn't correct about the redirect to step 4. When you do a POST on step-3 it will validate the form, set your new value and then as you have all the needed placeholders in the session, it will redirect you to the check notification page. `step-4` is never accessed.

I'll keep the behaviour anyway for consistency though.

Reply:
Have put in check for `== 0`. Note, flask already rejects negative numbers so `== 0` was fine rather than `<= 0`